app/main.py

```diff
@@ -16,4 +16,17 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+
+def sort_variables(list_object: list) -> dict:
+    my_sorted_variables = {"mutable": [], "immutable": []}
+    for variable_object in list_object:
+        if isinstance(variable_object, (int, str, bool, float, tuple)):
+            my_sorted_variables["immutable"].append(variable_object)
+        else:
+            my_sorted_variables["mutable"].append(variable_object)
+    return my_sorted_variables
+
+
+my_object = [lucky_number, pi, one_is_a_prime_number, name,
+             my_favourite_films, profile_info, marks, collection_of_coins]
+sorted_variables = sort_variables(my_object)
```

Comment:
In this task, you have overcomplicated everything; you just need to arrange everything manually. Here's an example:
```python
a = 123
b = []
c = "Hi!"
d = [1, 2]

sorted_variables = {
    "mutable": [b, d],
    "immutable": [a, c]
}
```